src/app/functions.js

```diff
@@ -0,0 +1,201 @@
+  const isLastTwoAreSame = sortedArray[3] === sortedArray[4]
+  const isFirstThreeAreSame = isFirstTwoAreSame && sortedArray[1] === sortedArray[2]
+  const isLastThreeAreSame = isLastTwoAreSame && sortedArray[2] === sortedArray[3]
+  if (isFirstThreeAreSame && isLastTwoAreSame && sortedArray[2] !== sortedArray[3]) {
+    return true
+  } else if (isFirstTwoAreSame && isLastThreeAreSame && sortedArray[1] !== sortedArray[2]) {
+    return true
+  } else {
+    return false
+  }
+}
+
+const fullHousePoints = numbers => {
+  return isFullHouse(numbers) ? 25 : 0
+}
+
+export const isSmallStraight = numbers => {
+  // 1-2-3-4
+  // 2-3-4-5
+  // 3-4-5-6
+  const sortedArray = numbers.sort((a, b) => a > b)
+  if (sortedArray.length > 5) return false
+  let consecutiveNumbers = 0
+  sortedArray.reduce((a, b) => {
+    if (a + 1 === b) consecutiveNumbers = consecutiveNumbers + 1
+    return b
+  })
+
+  return consecutiveNumbers === 3
+}
+
+const smallStraightPoints = numbers => {
+  return isSmallStraight(numbers) ? 30 : 0
+}
+
+export const isLargeStraight = numbers => {
+  // 1-2-3-4-5
+  // 2-3-4-5-6
+  const sortedArray = numbers.sort((a, b) => a > b)
+  if (sortedArray.length > 5) return false
+  let consecutiveNumbers = 0
+  sortedArray.reduce((a, b) => {
+    if (a + 1 === b) consecutiveNumbers = consecutiveNumbers + 1
+    return b
+  })
+
+  return consecutiveNumbers === 4
+}
+
+const largeStraightPoints = numbers => {
+  return isLargeStraight(numbers) ? 40 : 0
+}
+
+export const isYahtzee = numbers => {
+  const sortedArray = numbers.sort((a, b) => a > b)
+  if (sortedArray.length > 5) {
+    return false
+  }
+  let sameNumbers = 0
+  sortedArray.reduce((a, b) => {
+    if (a === b) sameNumbers = sameNumbers + 1
+    return b
+  })
+  return sameNumbers + 1 >= 5
+}
+
+const yahtzeePoints = numbers => {
+  return isYahtzee(numbers) ? 50 : 0
+}
+
+export const isChance = numbers => {
+  const sortedArray = numbers.sort((a, b) => a > b)
+  if (sortedArray.length > 5) return false
+  if (isFullHouse(numbers)) return true
+  let sameNumbers = 0
+  sortedArray.reduce((a, b) => {
+    if (a === b) sameNumbers = sameNumbers + 1
+    return b
+  })
+  return sameNumbers >= 1
+}
+
+const chancePoints = numbers => {
+  return isChance(numbers) ? sumAll(numbers) : 0
+}
+
+//
+// ** ALL GLOBAL FUNCTIONS THAT ARE BEING USED IN THE APP ** //
+//
+export const getRandomDiceNumber = () => {
+  return Math.floor(Math.random() * 6) + 1
+}
+
+export const getRandomNumberArray = () => {
+  return [
+    getRandomDiceNumber(),
+    getRandomDiceNumber(),
+    getRandomDiceNumber(),
+    getRandomDiceNumber(),
+    getRandomDiceNumber()
+  ]
+}
+
+export const getTotalPoints = numbers => {
+  return (
+    threeOfKindPoints(numbers) +
+    fourOfKindPoints(numbers) +
+    fullHousePoints(numbers) +
+    smallStraightPoints(numbers) +
+    largeStraightPoints(numbers) +
+    yahtzeePoints(numbers) +
+    chancePoints(numbers)
+  )
+}
```

Comment:
Also an option:
`sumAll(patterns.map(({ points }) => points(numbers)))`

It now stays relative to the pattern definition.